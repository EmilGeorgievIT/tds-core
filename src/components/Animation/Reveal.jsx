import React from 'react'
import PropTypes from 'prop-types'
import { Transition } from 'react-transition-group'

const defaultStyle = timeout => ({
  transition: `height ${timeout}ms ease-in-out`,
  height: 0,
  overflow: 'hidden',
})

const transitionStyles = {
  entered: { height: '200px' },
}

// TODO Fix the height! Can't do transition with "height: auto"

const Reveal = ({ children, ...rest }) => (
  <Transition {...rest}>
    {status => (
      <div
        style={{
          ...defaultStyle(rest.timeout),
          ...transitionStyles[status],
        }}
      >
        {children()}
      </div>
    )}
  </Transition>
)
Reveal.propTypes = {
  children: PropTypes.func.isRequired,
}

export default Reveal
