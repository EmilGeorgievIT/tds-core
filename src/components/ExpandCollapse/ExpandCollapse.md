```
<ExpandCollapse>
  <ExpandCollapse.Panel id="panel-1" header="First panel title">
    First panel
  </ExpandCollapse.Panel>
  <ExpandCollapse.Panel id="panel-2" header="Second panel title" subtext="Some subtext">
    Second panel
  </ExpandCollapse.Panel>
</ExpandCollapse>
```

### Arbitrary header component

```
<ExpandCollapse>
  <ExpandCollapse.Panel id="panel-10" header={<Text size="large">I'm large and misligned</Text>}>
    First panel
  </ExpandCollapse.Panel>
</ExpandCollapse>
```


### Adjusting height

```
initialState = {
  error: '',
};

const trigger = () => {
  if( state.error === '' ) {
    setState({error: 'Oh no there was an error'});
  }
  else {
    setState({error: ''});
  }
};

<ExpandCollapse open={['panel-21']}>
  <ExpandCollapse.Panel id="panel-20" header="First panel title">
    First panel
  </ExpandCollapse.Panel>
  <ExpandCollapse.Panel id="panel-21" header="Second panel title">
    <div>
      <Paragraph>Second panel blah blah blah</Paragraph>
      
      <Button onClick={trigger}>Trigger</Button>
      
      <Input label="Testing" error={state.error} />
    </div>
  </ExpandCollapse.Panel>
</ExpandCollapse>
```

### Disabled

```
<ExpandCollapse>
  <ExpandCollapse.Panel id="panel-30" header="First panel title" disabled>
    First panel
  </ExpandCollapse.Panel>
  <ExpandCollapse.Panel id="panel-31" header="Second panel title" subtext="Some subtext">
    Second panel
  </ExpandCollapse.Panel>
</ExpandCollapse>
```

### Accordions

```
const Accordion = require('./Accordion/Accordion').default;

<Accordion>
  <Accordion.Panel id="panel-40" header="First panel title">
    First panel
  </Accordion.Panel>
  <Accordion.Panel id="panel-41" header="Second panel title" subtext="Some subtext">
    Second panel
  </Accordion.Panel>
</Accordion>
```
