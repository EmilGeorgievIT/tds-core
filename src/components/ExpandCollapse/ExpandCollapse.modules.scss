@import '../../scss/settings/colours';

.base {
  background-color: $color-white;
}
