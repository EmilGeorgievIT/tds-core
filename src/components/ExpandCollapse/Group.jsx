import React, { Component, Children } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * A content area which can be collapsed and expanded.
 *
 * _This component can only be accessed as a name-spaced component: `ExpandCollapse.Group`._
 *
 * Group is used with `ExpandCollapse.Panel` to produce an ExpandCollapse set.
 * See [ExpandCollapse.Panel](#panel) for example.
 */
class Group extends Component {

  constructor(props) {
    super(props);
    const { activeKeys } = this.props;
    this.state = { currentActiveKeys: activeKeys };
  }

  componentWillReceiveProps(nextprops) {
    const nextActiveKeys = JSON.stringify(nextprops.activeKeys.sort());
    const currentActiveKeys = JSON.stringify(this.props.activeKeys.sort());

    if (nextActiveKeys !== currentActiveKeys) {
      this.setState({
        currentActiveKeys: nextprops.activeKeys
      });
    }
  }

  onPanelClick(panelKey) {
    return () => {
      if (this.props.disabledKeys.indexOf(panelKey) === -1) {
        const activeKeys = this.state.currentActiveKeys;
        if (activeKeys.indexOf(panelKey) > -1) {
          this.setState({
            currentActiveKeys: activeKeys.filter(activeKey => activeKey !== panelKey)
          });
        } else {
          /* eslint-disable no-lonely-if */
          if (this.props.accordion) {
            this.setState({ currentActiveKeys: [panelKey] });
          } else {
            this.setState({ currentActiveKeys: activeKeys.concat([panelKey]) });
          }
          /* eslint-enable no-lonely-if */
        }
        if (this.props.onChange) {
          this.props.onChange.call(this, panelKey);
        }
      }
    };
  }


  getPanels() {
    const activeKeys = this.state.currentActiveKeys;
    const newChildren = [];

    Children.forEach(this.props.children, (child, index) => {
      if (!child) return;
      // If there is no key provide, use the panel order as default key
      const panelKey = child.props.panelKey || String(index);
      const header = child.props.header;
      const isFirst = (index === 0);
      const isDisabled = this.props.disabledKeys.indexOf(panelKey) > -1;
      const isActive = activeKeys.indexOf(panelKey) > -1;

      const props = {
        key: index,
        panelKey,
        header,
        isActive,
        isDisabled,
        children: child.props.children,
        onPanelClick: this.onPanelClick(panelKey).bind(this),
        isFirst
      };
      newChildren.push(React.cloneElement(child, props));
    });

    return newChildren;
  }

  render() {
    const { className } = this.props;
    const collapseGroupClassName = classNames('collapsible-group', className);
    return (
      <div className={collapseGroupClassName}>{ this.getPanels() }</div>
    );
  }
}

Group.propTypes = {
  /**
   * One or more CSS class names separated by spaces to append onto the container.
   * Don't advertise as we plan on removing this feature soon.
   *
   * @ignore
   */
  className: PropTypes.string,
  /**
   * A key identifier that corresponds Group toggles with their respective expanded Panels.
   */
  activeKeys: PropTypes.array,
  /**
   * Similar to `activeKeys` except they correspond to collapsed Panels.
   */
  disabledKeys: PropTypes.array,
  /**
   * If true, sets Group behaviour to accordion.
   */
  accordion: PropTypes.bool,
  /**
   * The panels. Must be TDS ExpandCollapse.Panel components.
   */
  children: PropTypes.node,
  /**
   * A callback to invoke when a Panel is toggled.
   *
   * @param {Object} this The Group's context.
   * @param {String} panelKey The toggled Panel's key.
   */
  onChange: PropTypes.func
};

Group.defaultProps = {
  accordion: false,
  activeKeys: [],
  disabledKeys: []
};

export default Group;
