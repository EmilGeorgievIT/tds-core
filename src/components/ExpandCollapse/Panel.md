## Basic Usage

```
<Group disabledKeys={["panel-3"]}>
  <Panel header="Panel #1 Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus laoreet at lacus vel fringilla.">
    Panel #1 Body
  </Panel>
  <Panel header="Panel #2">
    Panel #2 Body
  </Panel>
  <Panel header="Panel #3 is disabled" panelKey="panel-3">
    Panel #3 Body
  </Panel>
</Group>
```
