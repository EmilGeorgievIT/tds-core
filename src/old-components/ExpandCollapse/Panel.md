## Basic usage

```
<Group>
  <Panel header="Specs">
    1.21 Gigawatts
  </Panel>
  <Panel header="Accessories">
    Charging cable, bumper case.
  </Panel>
  <Panel header="Warranty">
    1 Year, no questions asked.
  </Panel>
</Group>
```

## Accordion

Accordion is a special kind of Collapse, which allows only one panel to be expanded at a time.

```
<Group accordion>
  <Panel header="Specs">
    A fast CPU, a striking GPU.
  </Panel>
  <Panel header="Colours">
    Everything from tulips to crimsons.
  </Panel>
</Group>
```

## Disabling panels

```
<Group disabledKeys={["warranty"]}>
  <Panel header="Current Terms">
    Product is provided as-is.
  </Panel>
  <Panel header="Future Terms" panelKey="warranty">
    Under construction.
  </Panel>
</Group>
```

## Controlling the expanded panels externally

We can control the Expand/Collapse component state.

```
initialState = {
  activeKeys: []
};

const togglePanel = (panelKey) => {
  if (state.activeKeys.indexOf(panelKey) > -1) {
    setState({
      activeKeys: state.activeKeys.filter(k => k !== panelKey)
    });
  }
  else {
    setState({
      activeKeys: state.activeKeys.concat([panelKey])
    });
  }
};

<div>
  <div style={{marginBottom: 10}}>
    <Button variant="secondary" onClick={() => togglePanel('panel-1')}>
      Toggle panel #1
    </Button>
    <Button variant="secondary" onClick={() => togglePanel('panel-2')}>
      Toggle panel #2
    </Button>
  </div>

  <Group activeKeys={state.activeKeys}>
    <Panel header="Panel #1" panelKey="panel-1">
    Panel #1 Body
    </Panel>
    <Panel header="Panel #2" panelKey="panel-2">
    Panel #2 Body
    </Panel>
  </Group>
</div>
```
