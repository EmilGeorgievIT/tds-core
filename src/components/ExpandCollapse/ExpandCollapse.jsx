import React from 'react'
import PropTypes from 'prop-types'
import { childrenOfType } from 'airbnb-prop-types'
import Set from 'core-js/es6/set'

import safeRest from '../../utils/safeRest'
import { isEqual } from '../../utils/sets'

import PanelWrapper from './PanelWrapper/PanelWrapper'
import Panel from './Panel/Panel'

class ExpandCollapse extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      panels: new Set(props.open),
    }
  }

  componentWillReceiveProps(nextProps) {
    const nextPanels = new Set(nextProps.open)

    if (!isEqual(this.state.panels, nextPanels)) {
      this.setState({ panels: nextPanels })
    }
  }

  togglePanel = panelId => {
    const { onToggle } = this.props

    this.setState(
      ({ panels }) => {
        const nextPanels = new Set(panels)

        if (nextPanels.has(panelId)) {
          nextPanels.delete(panelId)
        } else {
          nextPanels.add(panelId)
        }

        return { panels: nextPanels }
      },
      () => {
        if (onToggle) {
          onToggle(this.state.panels)
        }
      }
    )
  }

  render() {
    const { children, ...rest } = this.props

    return (
      <div {...safeRest(rest)}>
        {React.Children.map(children, (panel, index) => {
          const { id, header, onToggle } = panel.props

          return (
            <PanelWrapper
              panelId={id}
              panelHeader={header}
              panelOnToggle={onToggle}
              open={this.state.panels.has(id)}
              last={index === React.Children.count(children) - 1}
              onClick={() => this.togglePanel(id)}
            >
              {panel}
            </PanelWrapper>
          )
        })}
      </div>
    )
  }
}

ExpandCollapse.propTypes = {
  open: PropTypes.array,
  onToggle: PropTypes.func,
  children: childrenOfType(Panel).isRequired,
}

ExpandCollapse.defaultProps = {
  open: [],
  onToggle: undefined,
}

ExpandCollapse.Panel = Panel

export default ExpandCollapse
