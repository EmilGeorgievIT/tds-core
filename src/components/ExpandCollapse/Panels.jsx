import React from 'react'
import PropTypes from 'prop-types'
import { childrenOfType } from 'airbnb-prop-types'

import safeRest from '../../utils/safeRest'

import HairlineDivider from '../Dividers/HairlineDivider/HairlineDivider'
import PanelWrapper from './PanelWrapper/PanelWrapper'
import Panel from './Panel/Panel'

import styles from './ExpandCollapse.modules.scss'

const getPanelWrappers = (isPanelOpen, togglePanel, children) =>
  React.Children.map(children, panel => {
    const { id, header, subtext, disabled, onToggle } = panel.props

    return (
      <PanelWrapper
        panelId={id}
        panelHeader={header}
        panelSubtext={subtext}
        panelOnToggle={onToggle}
        panelDisabled={disabled}
        open={isPanelOpen(id)}
        onClick={() => togglePanel(id)}
      >
        {panel}
      </PanelWrapper>
    )
  })

const wrapWithDividers = panels =>
  panels.map((panel, index) => {
    const last = index === panels.length - 1

    return [
      <HairlineDivider key={1} />,
      panel,
      last && <HairlineDivider key={2} />,
    ]
  })

const Panels = ({ isPanelOpen, togglePanel, children, ...rest }) => {
  const panels = getPanelWrappers(isPanelOpen, togglePanel, children)

  return (
    <div {...safeRest(rest)} className={styles.base}>
      {wrapWithDividers(panels)}
    </div>
  )
}

Panels.propTypes = {
  isPanelOpen: PropTypes.func.isRequired,
  togglePanel: PropTypes.func.isRequired,
  children: childrenOfType(Panel).isRequired,
}

export default Panels
