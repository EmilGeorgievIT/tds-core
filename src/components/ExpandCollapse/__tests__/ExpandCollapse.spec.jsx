import React from 'react'
import { shallow, mount } from 'enzyme'

import Reveal from '../../Animation/Reveal'
import ExpandCollapse from '../ExpandCollapse'

describe('ExpandCollapse', () => {
  const doShallow = (props = {}) =>
    shallow(
      <ExpandCollapse {...props}>
        <ExpandCollapse.Panel id="the-panel" header="The header">
          The panel content
        </ExpandCollapse.Panel>
      </ExpandCollapse>
    )

  const doMount = component => {
    const expandCollapse = mount(component)

    const findPanel = id => expandCollapse.find(`[data-testid="${id}"]`)

    return {
      expandCollapse,
      findPanel,
      togglePanel: id =>
        findPanel(id)
          .find('button')
          .simulate('click'),
    }
  }

  const expectPanelToBeOpen = panel => expect(panel.find(Reveal)).toHaveProp('in', true)
  const expectPanelToBeClosed = panel => expect(panel.find(Reveal)).toHaveProp('in', false)

  it('renders', () => {
    const { expandCollapse } = doMount(
      <ExpandCollapse open={['panel-1']}>
        <ExpandCollapse.Panel id="panel-1" header="First panel title">
          First panel
        </ExpandCollapse.Panel>
        <ExpandCollapse.Panel id="panel-2" header="Second panel title">
          Second panel
        </ExpandCollapse.Panel>
      </ExpandCollapse>
    )

    expect(expandCollapse).toMatchSnapshot()
  })

  it('can have some panels open and some panels closed by default', () => {
    const { findPanel } = doMount(
      <ExpandCollapse open={['panel-1']}>
        <ExpandCollapse.Panel id="panel-1" header="First panel title">
          First panel
        </ExpandCollapse.Panel>
        <ExpandCollapse.Panel id="panel-2" header="Second panel title">
          Second panel
        </ExpandCollapse.Panel>
      </ExpandCollapse>
    )

    expectPanelToBeOpen(findPanel('panel-1'))
    expectPanelToBeClosed(findPanel('panel-2'))
  })

  it('opens and closes panels by clicking them', () => {
    const { findPanel, togglePanel } = doMount(
      <ExpandCollapse>
        <ExpandCollapse.Panel id="panel-1" header="First panel title">
          First panel
        </ExpandCollapse.Panel>
      </ExpandCollapse>
    )

    togglePanel('panel-1')
    expectPanelToBeOpen(findPanel('panel-1'))

    togglePanel('panel-1')
    expectPanelToBeClosed(findPanel('panel-1'))
  })

  it('lets a parent component control the open and closed panels', () => {
    const { expandCollapse, findPanel } = doMount(
      <ExpandCollapse>
        <ExpandCollapse.Panel id="panel-1" header="First panel title">
          First panel
        </ExpandCollapse.Panel>
      </ExpandCollapse>
    )

    expandCollapse.setProps({ open: ['panel-1'] })
    expectPanelToBeOpen(findPanel('panel-1'))

    expandCollapse.setProps({ open: [] })
    expectPanelToBeClosed(findPanel('panel-1'))
  })

  it('triggers callbacks when panels are opened and closed', () => {
    const onPanelToggle = jest.fn()

    const { expandCollapse, togglePanel } = doMount(
      <ExpandCollapse>
        <ExpandCollapse.Panel id="panel-1" header="First panel title" onToggle={onPanelToggle}>
          First panel
        </ExpandCollapse.Panel>
      </ExpandCollapse>
    )

    togglePanel('panel-1')
    expect(onPanelToggle).toHaveBeenCalledWith(true)

    expandCollapse.setProps({ open: [] })
    expect(onPanelToggle).toHaveBeenCalledWith(false)

    // TODO: onToggle for the entire expand collapse that receives all the state
  })

  it('passes additional attributes to the element', () => {
    const expandCollapse = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(expandCollapse).toHaveProp('id', 'the-id')
    expect(expandCollapse).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const expandCollapse = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(expandCollapse).not.toHaveProp('className', 'my-custom-class')
    expect(expandCollapse).not.toHaveProp('style')
  })
})
