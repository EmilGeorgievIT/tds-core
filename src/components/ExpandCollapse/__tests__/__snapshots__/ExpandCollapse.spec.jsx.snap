// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandCollapse renders 1`] = `
<ExpandCollapse
  onToggle={undefined}
  open={
    Array [
      "panel-1",
    ]
  }
>
  <div
    className="base"
    onToggle={undefined}
    open={
      Array [
        "panel-1",
      ]
    }
  >
    <PanelWrapper
      key=".0"
      last={true}
      onClick={[Function]}
      open={true}
      panelDisabled={false}
      panelHeader="Panel title"
      panelId="panel-1"
      panelOnToggle={undefined}
      panelSubtext={undefined}
    >
      <div
        data-testid="panel-1"
      >
        <HairlineDivider
          gradient={false}
          vertical={false}
        >
          <hr
            className="horizontalThin"
          />
        </HairlineDivider>
        <Clickable
          dangerouslyAddClassName="header"
          dangerouslyAddStyle={undefined}
          disabled={false}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <button
            className="clickable header"
            disabled={false}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={undefined}
          >
            <Box
              all={undefined}
              bottom={undefined}
              dangerouslyAddClassName={undefined}
              horizontal={undefined}
              inline={false}
              left={undefined}
              right={undefined}
              spacing="padding"
              top={undefined}
              vertical={3}
            >
              <div
                className="verticalPadding-3"
              >
                <Flexbox
                  dangerouslyAddClassName={undefined}
                  direction="row"
                >
                  <div
                    className="row"
                  >
                    <Box
                      all={undefined}
                      bottom={undefined}
                      dangerouslyAddClassName={undefined}
                      horizontal={undefined}
                      inline={false}
                      left={undefined}
                      right={3}
                      spacing="margin"
                      top={undefined}
                      vertical={undefined}
                    >
                      <div
                        className="rightMargin-3"
                      >
                        <Translate
                          direction="y"
                          in={false}
                          length="0.25rem"
                          timeout={300}
                        >
                          <Transition
                            appear={false}
                            enter={true}
                            exit={true}
                            in={false}
                            mountOnEnter={false}
                            onEnter={[Function]}
                            onEntered={[Function]}
                            onEntering={[Function]}
                            onExit={[Function]}
                            onExited={[Function]}
                            onExiting={[Function]}
                            timeout={300}
                            unmountOnExit={false}
                          >
                            <div
                              style={
                                Object {
                                  "transition": "transform 300ms",
                                }
                              }
                            >
                              <DecorativeIcon
                                size={24}
                                symbol="caretDown"
                                variant="primary"
                              >
                                <Icon
                                  aria-hidden="true"
                                  size={24}
                                  symbol="caretDown"
                                  variant="primary"
                                >
                                  <i
                                    aria-hidden="true"
                                    className="iconCaretDown primary size24"
                                  />
                                </Icon>
                              </DecorativeIcon>
                            </div>
                          </Transition>
                        </Translate>
                      </div>
                    </Box>
                    <Flexbox
                      dangerouslyAddClassName={undefined}
                      direction="column"
                    >
                      <div
                        className="column"
                      >
                        <Box
                          all={undefined}
                          bottom={2}
                          dangerouslyAddClassName={undefined}
                          horizontal={undefined}
                          inline={false}
                          left={undefined}
                          right={undefined}
                          spacing="margin"
                          top={undefined}
                          vertical={undefined}
                        >
                          <div
                            className="bottomMargin-2"
                          >
                            <Text
                              block={false}
                              bold={false}
                              invert={false}
                              size="medium"
                            >
                              <span
                                className="medium mediumFont color"
                              >
                                Panel title
                              </span>
                            </Text>
                          </div>
                        </Box>
                      </div>
                    </Flexbox>
                  </div>
                </Flexbox>
              </div>
            </Box>
          </button>
        </Clickable>
        <Reveal
          height={0}
          in={true}
          timeout={500}
        >
          <Transition
            appear={false}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={500}
            unmountOnExit={false}
          >
            <div
              style={
                Object {
                  "height": "0px",
                  "overflow": "hidden",
                  "transition": "height 500ms",
                }
              }
            >
              <div>
                <DimpleDivider>
                  <hr
                    className="dimple"
                  />
                </DimpleDivider>
                <Box
                  all={undefined}
                  bottom={undefined}
                  dangerouslyAddClassName={undefined}
                  horizontal={undefined}
                  inline={false}
                  left={undefined}
                  right={undefined}
                  spacing="padding"
                  top={undefined}
                  vertical={3}
                >
                  <div
                    className="verticalPadding-3"
                  >
                    <Text
                      block={true}
                      bold={false}
                      invert={false}
                      size="base"
                    >
                      <div
                        className="base baseFont color"
                      >
                        <Panel
                          disabled={false}
                          header="Panel title"
                          id="panel-1"
                          onToggle={undefined}
                          subtext={undefined}
                        >
                          Panel content
                        </Panel>
                      </div>
                    </Text>
                  </div>
                </Box>
              </div>
            </div>
          </Transition>
        </Reveal>
        <HairlineDivider
          gradient={false}
          vertical={false}
        >
          <hr
            className="horizontalThin"
          />
        </HairlineDivider>
      </div>
    </PanelWrapper>
  </div>
</ExpandCollapse>
`;
